src/System.Text.Json/src/System/Text/Json/Serialization/JsonCamelCaseNamePolicy.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.Json.Serialization
+{
+    internal sealed class JsonCamelCaseNamePolicy : JsonNamingPolicy
+    {
+        public override string ConvertName(string name)
+        {
+            if (string.IsNullOrEmpty(name) || !char.IsUpper(name[0]))
+            {
+                return name;
+            }
+
+            char[] chars = name.ToCharArray();
+
+            for (int i = 0; i < chars.Length; i++)
+            {
+                if (i == 1 && !char.IsUpper(chars[i]))
+                {
+                    break;
+                }
+
+                bool hasNext = (i + 1 < chars.Length);
+
+                // Stop when next char is already lowercase.
+                if (i > 0 && hasNext && !char.IsUpper(chars[i + 1]))
```

Comment:
We use `ToUpper` here instead of `ToUpperInvariant`, but then use `ToLowerInvariant` to change the casing. Why?

Reply:
I assume you mean `IsUpper`. There is no `IsUpperInvariant` because the casing is already known by the Unicode table.